flax/experimental/nnx/README.md

```diff
@@ -96,407 +131,18 @@ def forward(static: nnx.ModuleDef, state: nnx.State, x: jax.Array):
-
-x = jnp.ones((3, 10))
-y = model(x, train=True, rngs=nnx.Rngs(dropout=1))
-```
-
-For a more robust implementation with comments take a look at the [Scan over layers](https://github.com/cgarciae/nnx/blob/main/examples/06_scan_over_layers.py) example.
-
-### Case Studies
-#### Shared State
-
-In NNX, you can create modules that share state between them. This is useful when designing complex neural network architectures, as it allows you to reuse certain layers and reduce the number of learnable parameters.
-
-Here's an example of creating a module with shared state:
-
-```python
-class Block(nnx.Module):
-    def __init__(self, linear: nnx.Linear, *, rngs: nnx.Rngs):
-        self.linear = linear
-        self.bn = nnx.BatchNorm(2, rngs=rngs)
-
-    def __call__(self, x, *, rngs: nnx.Rngs):
-        x = self.linear(x)
-        x = self.bn(x, rngs=rngs)
-        x = nnx.relu(x)
-        return x
-
-class Model(nnx.Module):
-    def __init__(self, *, rngs: nnx.Rngs):
-        shared = nnx.Linear(2, 2, rngs=rngs)
-        self.block1 = Block(shared, rngs=rngs)
-        self.block2 = Block(shared, rngs=rngs)
-
-    def __call__(self, x):
-        x = self.block1(x)
-        x = self.block2(x)
-        return x
-```
-
-In this example, the `Model` module contains two instances of the `Block` module. Each instance shares the same `nnx.Linear` module. To run the model, you can use the Rngs `flags` argument to set the `use_running_average` flag for all `BatchNorm` modules.
-
-Here's an example of computing the loss for a `Model` instance:
-
-```python
-def loss_fn(model: Model, x: jax.Array, y: jax.Array):
-    with nnx.flags(use_running_average=True):
-        y_pred = model(x)
-    return jnp.mean((y - y_pred) ** 2)
-```
+* [LM1B](https://github.com/google/flax/tree/main/flax/experimental/nnx/examples/lm1b): A language model trained on the 1 Billion Word Benchmark dataset.
 
-It's important to note that the state for the shared `nnx.Linear` module will be kept in sync at all times on both `Block` instances, including during gradient updates.
+#### Toy Examples
+* [Using the Functional API](https://github.com/google/flax/tree/main/flax/experimental/nnx/examples/toy_examples/01_functional_api.py): Shows how to train a simple model using the functional API.
+* [Using Lifted Transforms](https://github.com/google/flax/tree/main/flax/experimental/nnx/examples/toy_examples/02_lifted_transforms.py): Shows how to train a simple model using lifted transforms.
+* [Using TrainState](https://github.com/google/flax/tree/main/flax/experimental/nnx/examples/toy_examples/03_train_state.py): Shows how to train a simple model using the functional API with the help of `TrainState`.
+* [Training a VAE](https://github.com/google/flax/tree/main/flax/experimental/nnx/examples/toy_examples/05_vae.py): Shows how to train a VAE on the binarized MNIST dataset, uses the functional API, `TrainState`, and shows how to use capture intermediate values to retrieve `kl_loss`.
```

Comment:
Should `05_vae.py` be renamed to `04_vae.py` (as well as decrementing all subsequent toy example numbers)? The number `04` seems to be missing from the [`toy_examples`](https://github.com/google/flax/tree/main/flax/experimental/nnx/examples/toy_examples).

Reply:
Yeah there is a bit of an inconsistency here. I don't think its super important but we sould fix it at some point :)